R/learn.R

```diff
@@ -192,16 +192,15 @@ legacy_new_dm <- function(tables, data_model) {
   segment <- data_model_tables$segment
   # would be logical NA otherwise, but if set, it is class `character`
   display <- as.character(data_model_tables$display)
-  filter <- new_filters()
   zoom <- new_zoom()
   key_tracker_zoom <- new_key_tracker_zoom()
 
-  # Legacy compatibility
-  pks$column <- as.list(pks$column)
-
   pks <-
     pks %>%
-    nest(pks = -table)
+    # Legacy compatibility
+    mutate(column = vctrs::vec_cast(column, list())) %>%
+    nest(pks = -table) %>%
+    mutate(pks = vctrs::as_list_of(pks))
```

Comment:
This won't be necessary with https://github.com/tidyverse/tidyr/pull/847 I think.

Reply:
...but it's going to take a while. Can you please create a small wrapper `nest_compat()` and add an issue?